### PR TITLE
Fix: Characters split on paste

### LINKED ISSUE #133: Paste: Do not split on special characters
When a user pastes HTML (which calls [importText](https://github.com/cybersemics/em/blob/b4fd8faa535eb423fdb620d4f22040ff8129634c/src/util.js#L1354)), `#` and `&` characters (and perhaps others) are incorrectly split into multiple lines. 

The fix must not break copy-and-paste from raw HTML or from formatted text (such as a bulleted list), and from single line or multiple lines, as described in detail below.

## Expected behavior

- If the user copies a single line of plain text that contains a `&` character, pasting into **em** should create a single thought.
- If the user copies a single line of HTML that contains `&amp;` (or any other character entity), pasting into **em** should create a single thought.
- If the user copies three lines of plain text that contain a `&` character, pasting into **em** should create three thoughts.
- If the user copies three lines of HTML (e.g. three `<li>`) that contain `&amp;` (or any other character entity), pasting into **em** should create three thoughts.

### Sample text/HTML to copy:

- A
- B
- C & D

```html
<li>A</li>
<li>B</li>
<li>C &amp; D</li>
```

### PR DESCRIPTION
**Pull request by mul53**
Closes #133

### CODE REVIEW

File: src/components/Editable.js [hunk 1]
```diff
@@ -236,16 +238,21 @@ export const Editable = connect()(({ focus, itemsRanked, contextChain, showConte
       // the data will be available as text/plain or text/html
       // this reflects the format of the source data more than the actual contents
       // text/plain may contain text that ultimately looks like html (contains <li>) and should be parsed as html
-      const plainText = e.clipboardData.getData('text/plain')
-      const htmlText = e.clipboardData.getData('text/html')
+      let plainText = e.clipboardData.getData('text/plain')
+      let htmlText = e.clipboardData.getData('text/html')
```
Comment (raineorshine) [inline src/components/Editable.js]: Please read the [Contributing Guidelines](https://github.com/cybersemics/em/blob/dev/CONTRIBUTING.md). `let` is not allowed on this project. You must create a new variable and use `const`.

File: src/util.js [hunk 2]
```diff
@@ -1909,3 +1909,25 @@ export const sync = (dataUpdates={}, contextChildrenUpdates={}, { localOnly, for
...
+  const rules = { '(&amp;|&)': 'and' }
+  let res = str
+
+  for (let rule of Object.keys(rules)) {
+    const regex = new RegExp(rule, 'gi')
+    res = res.replace(regex, rules[rule])
+  }
+
+  return res
+}
```
Comment (raineorshine) [inline src/util.js]: Again note the Contributing Guidelines. `let` and `for` are not allowed on this project.

File: src/util.js [hunk 3]
```diff
@@ -1909,3 +1909,25 @@ export const sync = (dataUpdates={}, contextChildrenUpdates={}, { localOnly, for
...
+ * @param {String} str 
+ */
+export const isHTML = (str) => /<\/?[a-z][\s\S]*>/i.test(str);
+
+/**
+ * Replaces certain reserved HTML chars
+ * @param {String} str 
+ */
+export const replaceSpecialHTMLChars = (str) => {
+  const rules = { '(&amp;|&)': 'and' }
```
Comment (raineorshine) [inline src/util.js]: The task does not ask for `&amp;` to be replaced by `and`. Rather, `&` should be allowed and rendered appropriately for the user.

**Comment by mul53** [comment]
I took a different approach on this PR, i hope it fits the requirements

**Comment by raineorshine** [comment]
Thanks, I will clone it and try it out!

**Comment by raineorshine** [comment]
Well done, this works great! That's totally the behavior I want. Just two other issues.

1. Can you strip out any HTML from `sigKey` in [Link](https://github.com/cybersemics/em/blob/dev/src/components/Link.js) so that the breadcrumbs appear as plain text?

Screen Shot 2019-11-21 at 10 22 13 AM

2. When the user types `<li>Some Text</li>` and then deletes the next sibling thought, the following error occurs. I realize this is a little more involved. If this is a quick fix, that would be great, but please don't hesitate to let me know if it's out of scope!

Screen Shot 2019-11-21 at 10 25 01 AM

**Comment by mul53** [comment]
1)  I have stripped the HTML, it now looks much better
2) Hmmmmm it seems a bit involving after looking at it for a while

And it is possible if i can sign up on the bounty for this issue?

**Comment by raineorshine** [comment]
Thanks!

**Pull request merged**